File: problems-extend/3701-3800/3759_count-elements-with-at-least-k-greater-values/solution_sort_suffix.rs
impl Solution {
    pub fn count_elements(nums: Vec<i32>, k: i32) -> i32 {
        // Sorting lines every element up with its rank: the elements
        // strictly greater than a value are exactly the sorted suffix
        // after that value's run. The whole count hangs on one threshold,
        // the value at sorted index t = n - k - 1.
        let mut ordered = nums;
        ordered.sort_unstable();
        let n = ordered.len();
        let threshold = ordered[n - 1 - k as usize];
        // Elements strictly below the threshold all qualify: their runs
        // end before it. The run AT the threshold qualifies only when its
        // last member still sees >= k strictly greater values, i.e. the
        // run ends at or before t. Values above the threshold never do.
        let left = ordered.partition_point(|&v| v < threshold);
        let right = ordered.partition_point(|&v| v <= threshold);
        if (n - right) as i32 >= k {
            return right as i32;
        }
        left as i32
    }
}

File: problems-originals/3701-3800/3759_count-elements-with-at-least-k-greater-values/solution_sort_suffix.ts
function countElements(nums: number[], k: number): number {
    // Sorting lines every element up with its rank: the elements strictly
    // greater than a value are exactly the sorted suffix after that
    // value's run. The whole count hangs on one threshold, the value at
    // sorted index t = n - k - 1.
    const ordered = [...nums].sort((a, b) => a - b);
    const n = ordered.length;
    const threshold = ordered[n - k - 1];
    // Elements strictly below the threshold all qualify: their runs end
    // before it. The run AT the threshold qualifies only when its last
    // member still sees >= k strictly greater values, i.e. the run ends
    // at or before t. Values above the threshold never qualify.
    let left = 0;
    while (ordered[left] < threshold) left++;
    let right = left;
    while (right < n && ordered[right] === threshold) right++;
    return n - right >= k ? right : left;
}
